### PR TITLE
[EISW-] [master] Rename `VPU` to `NPU` in openvino codebase

### PR DESCRIPTION
**Pull request by MirceaDan99**
### Details:
 - *item1*
 - *...*
 - PR for `releases/2023/0`: https://github.com/openvinotoolkit/openvino/pull/18974

### Tickets:
 - *ticket-id*

### CODE REVIEW

**Comment by MirceaDan99** [comment]
PR for `releases/2023/0`: https://github.com/openvinotoolkit/openvino/pull/18974

File: src/inference/include/ie/vpu/myriad_config.hpp [hunk 1]
```diff
@@ -60,7 +60,7 @@ INFERENCE_ENGINE_1_0_DEPRECATED DECLARE_VPU_CONFIG(MYRIAD_PCIE);
 INFERENCE_ENGINE_1_0_DEPRECATED DECLARE_VPU_CONFIG(MYRIAD_USB);
 
 /**
- * @brief Optimize vpu plugin execution to maximize throughput.
+ * @brief Optimize npu plugin execution to maximize throughput.
```
Comment (ArtemySkrebkov) [inline src/inference/include/ie/vpu/myriad_config.hpp:63]: Often, when openvino code base mentions VPU, it actually means MYRIAD plugin. Is it supposed to be renamed to NPU as well?

Comment (DariaMityagina) [reply src/inference/include/ie/vpu/myriad_config.hpp:63]: As far as I know the Myriad plugin was recently removed. It still remains only in the `releases/2022/3` branch.

Comment (ilya-lavrenov) [reply src/inference/include/ie/vpu/myriad_config.hpp:63]: let's rename into `MYRIAD` in this place

File: tools/pot/openvino/tools/pot/graph/utils.py [hunk 2]
```diff
@@ -19,7 +19,7 @@
     'ANY': get_cpu_ignored_patterns(),
     'CPU': get_cpu_ignored_patterns(),
     'GPU': get_gpu_ignored_patterns(),
-    'VPU': get_vpu_ignored_patterns(),
+    'NPU': get_vpu_ignored_patterns(),
```
Comment (ilya-lavrenov) [inline tools/pot/openvino/tools/pot/graph/utils.py:22]:
```suggestion
    'NPU': get_npu_ignored_patterns(),
```

**Review by ilya-lavrenov — APPROVED** [review]

**Review by rzubarev — APPROVED** [review]

File: src/tests/test_utils/common_test_utils/include/common_test_utils/test_common.hpp [hunk 3]
```diff
@@ -18,7 +18,7 @@ class TestsCommon;
 
 // Temporal WA for applications.ai.vpu-accelerators.vpux-plugin repo
```
Comment (wangleis) [inline src/tests/test_utils/common_test_utils/include/common_test_utils/test_common.hpp:19]: Please update `vpu` and `vpux` in comments as well

File: tests/time_tests/test_runner/test_timetest.py [hunk 4]
```diff
@@ -43,7 +43,7 @@ def test_timetest(instance, executable, niter, cl_cache_dir, model_cache, model_
     :param niter: number of times to run executable
     :param cl_cache_dir: directory to store OpenCL cache
     :param cpu_cache: flag to enable model CPU cache
-    :param vpu_compiler: flag to change VPU compiler type
+    :param vpu_compiler: flag to change NPU compiler type
```
Comment (wangleis) [inline tests/time_tests/test_runner/test_timetest.py:46]: vpu_compiler -> npu_compiler

**Comment by ilya-lavrenov** [comment]
build_jenkins

File: src/inference/include/ie/vpu/myriad_config.hpp [hunk 5]
```diff
@@ -60,7 +60,7 @@ INFERENCE_ENGINE_1_0_DEPRECATED DECLARE_VPU_CONFIG(MYRIAD_PCIE);
 INFERENCE_ENGINE_1_0_DEPRECATED DECLARE_VPU_CONFIG(MYRIAD_USB);
```
Comment (wangleis) [inline src/inference/include/ie/vpu/myriad_config.hpp:60]: Please update `vpu` in folder name and definition

File: src/inference/include/ie/vpu/vpu_config.hpp [hunk 6]
```diff
@@ -3,7 +3,7 @@
 //
 
 /**
- * @brief A header that defines common config subset for VPU plugins.
+ * @brief A header that defines common config subset for NPU plugins.
  * Include myriad_config.hpp or hddl_config.hpp directly.
  * These properties should be used in SetConfig() and LoadNetwork() methods of plugins
  *
```
Comment (wangleis) [inline src/inference/include/ie/vpu/vpu_config.hpp:9]: Please update `vpu_config.hpp` in comments

File: tools/pot/openvino/tools/pot/algorithms/quantization/README.md [hunk 7]
```diff
@@ -4,7 +4,7 @@
 
 The primary optimization feature of the Post-training Optimization Tool (POT) is the uniform integer quantization which allows substantially increasing inference performance and reducing the model size. Different HW platforms can support different integer precisions and POT is designed to support all of them, for example, 8-bit for CPU, GPU, VPU, 16-bit for GNA. Moreover, POT makes the specification of HW settings transparent for the user by introducing a concept of the `target_device` parameter.
```
Comment (wangleis) [inline tools/pot/openvino/tools/pot/algorithms/quantization/README.md:5]: `8-bit for CPU, GPU, VPU` -> `8-bit for CPU, GPU, NPU`

File: tools/pot/openvino/tools/pot/algorithms/quantization/utils.py [hunk 8]
```diff
@@ -21,7 +21,7 @@
                           'GNA3': 'gna3.json',
                           'GNA3.5': 'gna3.json',
                           'GPU': 'gpu.json',  # Same as cpu.json but without LSTM/GRUSequence quantization
-                          'VPU': 'vpu.json',
+                          'NPU': 'vpu.json',
```
Comment (wangleis) [inline tools/pot/openvino/tools/pot/algorithms/quantization/utils.py:24]: 'vpu.json' ->'npu.json'

File: tools/pot/openvino/tools/pot/graph/utils.py [hunk 9]
```diff
@@ -10,7 +10,7 @@
 from openvino.tools.pot.version import get_version
 from .cpu_patterns import get_cpu_ignored_patterns, get_cpu_spr_ignored_patterns
 from .gpu_patterns import get_gpu_ignored_patterns
-from .vpu_patterns import get_vpu_ignored_patterns
+from .vpu_patterns import get_npu_ignored_patterns
```
Comment (wangleis) [inline tools/pot/openvino/tools/pot/graph/utils.py:13]: vpu_patterns

**Review by ArtemySkrebkov — APPROVED** [review]

File: src/tests/test_utils/common_test_utils/include/common_test_utils/test_common.hpp (hunk 3 above)
Comment (PatrikStepan) [reply src/tests/test_utils/common_test_utils/include/common_test_utils/test_common.hpp:19]: This specific comment refers to the repo name - we will not be able to update the repo name in this round of renaming. Will search again for other comments.

File: tests/time_tests/test_runner/test_timetest.py (hunk 4 above)
Comment (PatrikStepan) [reply tests/time_tests/test_runner/test_timetest.py]: Should we only update public UI in this PR ( device names, library names)? That was the intention.
I am afraid this parameter is used by other scripts that will break. 
We will later gradually remove VPU from both our code and OV code.

File: docs/resources/Supported_Devices.md [hunk 10]
```diff
@@ -14,7 +14,7 @@ Currently, processors of the 11th generation and later (up to the 13th generatio
 
 .. note::
 
-   With OpenVINO™ 2023.0 release, support has been cancelled for all VPU accelerators based on Intel® Movidius™.
+   With OpenVINO™ 2023.0 release, support has been cancelled for all NPU accelerators based on Intel® Movidius™.
```
Comment (PatrikStepan) [inline docs/resources/Supported_Devices.md:17]: I think this refers to the old VPU devices. Should we keep VPU here for clarity?

**Comment by PatrikStepan** [comment]
@ilya-lavrenov , @wangleis  should we keep everything that refers to hddl or myriadX still named as VPU? The intention and ask from our program is to rename the current CCG products to NPU ( Neural Processor Unit).

**Comment by PatrikStepan** [comment]
> > @ilya-lavrenov , @wangleis should we keep everything that refers to hddl or myriadX still named as VPU? The intention and ask from our program is to rename the current CCG products to NPU ( Neural Processor Unit).
> 
> Myriad and HDDL must be removed at all, because they are not part of OpenVINO anymore

Will remove this then:
image

**Review by wangleis — APPROVED** [review]

File: src/tests/test_utils/common_test_utils/src/test_constants.cpp [hunk 11]
```diff
@@ -13,7 +13,7 @@ const char *DEVICE_CPU = "CPU";
 const char *DEVICE_GNA = "GNA";
 const char *DEVICE_GPU = "GPU";
 const char *DEVICE_BATCH = "BATCH";
-const char *DEVICE_KEEMBAY = "VPU";
+const char *DEVICE_KEEMBAY = "NPU";
```
Comment (PatrikStepan) [inline src/tests/test_utils/common_test_utils/src/test_constants.cpp:16]: Can we rename this to DEVICE_NPU and move it before BATCH? Few other occurrences of this define in OV codebase and functional tests need to be updated as well in vpux-plugin repo.

**Comment by PatrikStepan** [comment]
> > > @ilya-lavrenov , @wangleis should we keep everything that refers to hddl or myriadX still named as VPU? The intention and ask from our program is to rename the current CCG products to NPU ( Neural Processor Unit).
> > 
> > 
> > Myriad and HDDL must be removed at all, because they are not part of OpenVINO anymore
> 
> Will remove this then: image

We cannot remove this folder in this PR unfortunately, I see that it is still used in some functional test. We should rather not modify the content of this folder at all and remove it in another PR.

File: src/tests/test_utils/common_test_utils/src/test_constants.cpp (hunk 11 above)
Comment (StefaniaHergane) [reply src/tests/test_utils/common_test_utils/src/test_constants.cpp:16]: Updated in PR https://github.com/openvinotoolkit/openvino/pull/19004 which will continue this thread

**Comment by StefaniaHergane** [comment]
Change requests will be further addressed in: https://github.com/openvinotoolkit/openvino/pull/19004
Unfortunately, we will have to continue on that PR and close this one (author will be OOO and not able to address reviews).

@ilya-lavrenov , can you please close this as a maintainer so we can continue on the other PR?
Thank you!

**Pull request closed without merging**